packages/kbn-search-api-panels/components/ingest_pipeline_panel.tsx

```diff
@@ -0,0 +1,131 @@
+ * Side Public License, v 1.
+ */
+
+import React, { useState, Fragment, useMemo } from 'react';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  EuiBadge,
+  EuiSuperSelect,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import type { Pipeline } from '@kbn/ingest-pipelines-plugin/public';
+
+interface OptionItem {
+  value: string;
+  inputDisplay: string;
+  dropdownDisplay: JSX.Element;
+}
+
+interface IngestPipelinePanelProps {
+  setSelectedPipeline: (pipeline: string) => void;
+  ingestPipelineData?: Pipeline[] | null;
+}
+
+const RECOMMENDED_INGEST_PIPELINE = 'search-default-ingestion';
+
+export const IngestPipelinePanel: React.FC<IngestPipelinePanelProps> = ({
+  setSelectedPipeline,
+  ingestPipelineData,
+}) => {
+  const options = useMemo(() => {
+    return ingestPipelineData
+      ? ingestPipelineData.map(
+          (item): OptionItem => ({
+            value: item.name,
+            inputDisplay: item.name,
+            dropdownDisplay: (
+              <Fragment>
+                <strong>{item.name}</strong>
+                <EuiFlexGroup
+                  gutterSize="s"
+                  alignItems="center"
+                  data-test-subj="ingestPipelinePanelProcessors"
+                >
+                  <EuiFlexItem grow={false}>
+                    <EuiText size="s" color="subdued">
+                      <p>{`${item.processors.length} ${
+                        item.processors.length > 1 ? 'processors' : 'processor'
+                      }`}</p>
+                    </EuiText>
+                  </EuiFlexItem>
+                  {item.isManaged && (
+                    <EuiFlexItem grow={false}>
+                      <EuiBadge>Managed</EuiBadge>
+                    </EuiFlexItem>
+                  )}
+                  {item.name === RECOMMENDED_INGEST_PIPELINE && (
+                    <EuiFlexItem grow={false}>
+                      <EuiBadge color="primary">Recommended</EuiBadge>
+                    </EuiFlexItem>
+                  )}
+                </EuiFlexGroup>
+              </Fragment>
+            ),
+          })
+        )
+      : [];
+  }, [ingestPipelineData]);
+
+  const [selected, setSelected] = useState<string>();
+
+  const onChange = (value: string) => {
+    setSelected(value);
+    setSelectedPipeline(value);
+  };
+
+  return (
+    <>
+      <EuiFlexGroup gutterSize="s" data-test-subj="ingestPipelinePanelTitle">
+        <EuiFlexItem grow={false}>
+          <EuiTitle size="xs">
+            <strong>
+              {i18n.translate(
+                'xpack.serverlessSearch.indexManagement.indexDetails.pipelinePanel.title',
+                {
+                  defaultMessage: 'Preprocess your data',
+                }
+              )}
+            </strong>
+          </EuiTitle>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiBadge>Optional</EuiBadge>
```

Comment:
should be translated as well

Reply:
there is constant variable for [optional](https://github.com/elastic/kibana/blob/0c0cf0a8e0e9dde31f9c9872cde328984065dfc8/x-pack/plugins/serverless_search/common/i18n_string.ts#L42) which can be used

Reply:
Just like this [comment](https://github.com/elastic/kibana/pull/182604#discussion_r1594309571), we may need to move this to @kbn-search-api-panels.

Reply:
Please ignore this comment. We can rather define a new translation than dealing with dependency problems.

IMO Ideally, we can have constant variable with translation for Optional, Recommended string in `kbn-search-api-panels `similar to what we have for [optional](https://github.com/elastic/kibana/blob/0c0cf0a8e0e9dde31f9c9872cde328984065dfc8/x-pack/plugins/serverless_search/common/i18n_string.ts#L42) in serverless_search plugin. But that can be a different PR.